### ISSUE TITLE
Doesn't generate

### ISSUE THREAD

**Issue by Delitants** [K-vanc/Tempest-EPG-Generator#21]
Doesn't generate anything, XML came out with blank values with no programming. Except for ion.
Dead siteconfig?

![Screenshot 2023-03-07 at 2 36 03 PM](https://user-images.githubusercontent.com/2836790/223588569-401c0c95-d4b8-448f-8b9e-275a3c3d4f53.png)
![Screenshot 2023-03-07 at 2 37 01 PM](https://user-images.githubusercontent.com/2836790/223588673-fb1cfd64-0292-4d77-89dd-18d9a639f277.png)

**Comment by Delitants** [comment]
(1/55) Site: ionplustv.com_0
Channel id: ion_plus
Xmltv id: ION Plus
28 shows grabbed in 0.08 seconds

(2/55) Site: [ENC][EX]whattowatch.com[US]_0
Channel id: 6219
Xmltv id: [RCN Cable - New York City, NY - Digital] AMC - Eastern Feed HD / AMC-HD
**No show grabbed in 0.00 seconds**

(3/55) Site: [ENC][EX]whattowatch.com[US]_0
Channel id: 10876
Xmltv id: [Dish - National, NY ] Outside Television / OUT-TV
**No show grabbed in 0.00 seconds**

**Comment by K-vanc** [comment]
it is not dead siteconfig and working but probably, they started to apply some kind of geo-blocking rule. I will check deeply but I am able to grab with several timeouts. I tested some on another machine with network setting and worked without any timeout error
![image](https://user-images.githubusercontent.com/97025515/223621006-1f04cc0c-4efc-43a3-82bb-12d6d06cf911.png)

**Comment by K-vanc** [comment]
I recreated channel list with New York zip code again set new channels. Increased timeout value to 20 in my configuration file. It is running without any error and seems no geo-blocking etc.

![image](https://user-images.githubusercontent.com/97025515/223623118-db796223-2995-4257-bf64-e62130ab53cd.png)

Maybe your ip is blocked by them. Many website has flood protection and if you make toooooo many repeating trial request in a short time, they may temporarily black list you. It is better to  reset your modem, wait a bit and try again. If you have vpn, you may try running it in the same machine which hosts Tempest, to test.

**Comment by K-vanc** [comment]
BTW, as usual, noone is mentioning this very useless details since writting "not working" much more effective but could you please reply below questions;

Tempest Version:
Php Version:
Operating System:
Webserver Name:

**Comment by Delitants** [comment]
> Maybe your ip is blocked by them. Many website has flood protection and if you make toooooo many repeating trial request in a short time, they may temporarily black list you. It is better to reset your modem, wait a bit and try again. If you have vpn, you may try running it in the same machine which hosts Tempest, to test.

I ran it literally once-twice. There is no vpn, but hosting.

> BTW, as usual, noone is mentioning this very useless details since writting "not working" much more effective but could you please reply below questions;

Well I'm sorry, but your error logging option is all it shows posted above. Not so informative.

> Tempest Version: Php Version: Operating System: Webserver Name:
> 
> and are you able to grab from only [ENC] prefixed siteconfigs? I just noticed that also you cannot grab from tvpassport and return times are 0.00 second which means, there is problem with your server setup and operation immediately terminated for [ENC][EX] siteconfigs

rv1.5.0
7.4 PHP
Centos 7 (Cloudlinux)
Apache LSAPI

There is absolutely no php errors in log, everything is set up properly.

None of the ENC seems working. Tried directtv, tvpassport, whattowatch. All give 0.00, only ion and ion plus do work.

**Comment by K-vanc** [comment]
ok, thanks for the info. Tvpassport is currently out-of-service with a message on their website for couple of days. It is not about ip block or it cannot be solved via vpn. There are 2 possibilities;

* A few weeks ago, another user mentioned similar issue and solved after installation of some missing modules. Normally, he  was using archlinux and Tempest has a built-in module checker but somehow, it couldnt catch his missing modules and I cannot reproduce the issue to fix since it is catching both in my windows and debian and I dont have an environment to test in archlinux. Your case may be similar on CentOS. if yes, I will start focusing module name differences. Since you are able to run unlocked ones but not [ENC] ones, it could be related with openssl;
https://github.com/K-vanc/Tempest-EPG-Generator/issues/19#issuecomment-1440603937

* You mentioned about Cloud hosting service. I have a few notice as some hosting services are not allowing Tempest to run properly especially on [ENC] siteconfig files. Previously, I uploaded a demo version to a free hosting service and it was working with [ENC] siteconfigs. A few user sent me their username/password to check their hosting settings/permission/ownership etc. and despite everything was set properly and all modules were looking installed, probably related with some security setting of service, [ENC] siteconfigs were terminated immediately with 0.00 seconds. I tried several things but nothing worked. If this is your case, unfortunately, there is nothing I can do. Possible reasons could be(I am not %100 sure), blocked udp port/socket opening and/or blocked access into system's temporary folder. Maybe, you can ask your service provider to check

**Comment by Delitants** [comment]
If there is any modules missing, why doesn't it report it in error logs?
My version isn't 8.2.2 at all, openssl is installed. Hosting isn't free at all, it's a highly customizable container, I can enable any modules or change settings of the php I want. Just need to know what.
I did check opening mentioned tvpassport from hosting's IP in brower and nothing is blocked there, I do see programming.
Tvpassport's out of service does not explain why directtv and other also terminated with 0.00.

Where to enable a detailed debug tho? Or we'll just keep guessing around.

**Comment by Delitants** [comment]
Here is modules of currently running php, you can see what's on or off. I can turn on anything listed.
[PHP_extension for CGI and LSAPI - 7.4.33 (alt).pdf](https://github.com/K-vanc/Tempest-EPG-Generator/files/10918206/PHP_extension.for.CGI.and.LSAPI.-.7.4.33.alt.pdf)

**Comment by Delitants** [comment]
Lit lamp bulb icon is enabled, dark bulb - disabled, rest icons is irrelevant.

**Comment by K-vanc** [comment]
yes, modules seem ok. Did you able to check log after error reporting?

**Comment by Delitants** [comment]
> yes, modules seem ok. Did you able to check log after error reporting?

I did

`<?php
error_reporting(1);
/*     Tempest EPG Generator (made by Kvanc)
https://github.com/K-vanc/Tempest-EPG-Generator.git  */`

Produces no errors in log after I click generate epg.

**Comment by K-vanc** [comment]
It is same behaviour as I explained from other users hosting service. Something is blocked by service, maybe some module enabled but partially working.

You have already enabled verbose. If it would be an error, you could see it on error log now. It is not siteconfig problem, i am able to grab with attached screenshoots and your server is not starting to grab on these siteconfigs with "0.00 seconds" termination time. 

If you set up a local machine, you will see same files are working. I dont know why some hosting service are running while some of them causing identical termination But unfortunately, i cannot change their security policy.

I received below screenshot from another user who was having same issue on a hosting service from Czech Rep. I logged in to his account and checked/changed every single setting but nothing changed as always terminated 0.00 seconds then installed Xammp to his laptop and worked same files.

![E9DD5287-E4D7-4CF9-A422-75B65B1F6F20.png](https://user-images.githubusercontent.com/97025515/223804448-9ef9ae30-46a5-4f47-9476-30d4a24b79ef.png)

**Comment by K-vanc** [comment]
> You can't blame hosting if you can't identify the problem. It would be as easy as write to support, but I have nothing to report. Don't expect everyone use their laptops for your software, this is no more than just a sandbox, and I def don't want all the web server to be sitting on my laptop.

I am not blaming anyone. I am explaning what it is happening and also explaning "I cannot help for something I cannot change..". I already explained you hot topics that you can ask your support if you Scroll up but instead of you are choosing to blaim me..

**Comment by K-vanc** [comment]
> Verbose output has nothing to do with error log.
> I expect to see a step-by-step reporting on the page.
> For example:
> 1. Connecting to tvpassport.com
> 2. Connection established, answer is 200 OK
> 3. Fetching channel A
> 4. Error remote server responded with 403. Your IP is blocked?
> ....
> and so on.
> Very easy and informative.

You may see this and more when you run an unlocked siteconfig on debugger for I already explained what is your case as;

* not connecting to tvpassport
* because your server is not allowing it to connect
* actually your server just reading [ENC] siteconfig file and stops without making any request
* thats why it is ending in 0.00 seconds
 
Unfortunately, i will not give you what you are asking so please use another handy and very useful tool instead of Tempest

**Comment by K-vanc** [comment]
> Huh. How funny. You confirmed yourself that your software doesn’t run on any hosting, including yours in CZ. And closed the issue as wontfix. I mean… this is hilarious approach. Very professional. Yup, it’s solely my problem, of course. Good luck!

Yes, i confirm it is not working at all so please use another solution. That hosting was not mine since I have never been in CZ but as I explained didnt work on it as well. I only tested on webhost000 as free service and it worked months ago but if I try again now, for sure will not work. So please do not waste your time here. I clearly described which type of users needed for Tempest and you are not one of mentioned in my Declaration;

https://github.com/K-vanc/Tempest-EPG-Generator/blob/main/DECLARATION.md
